kernel/entry/common.c

```diff
@@ -26,6 +27,7 @@ static __always_inline void enter_from_user_mode(struct pt_regs *regs)
 	CT_WARN_ON(ct_state() != CONTEXT_USER);
 	user_exit_irqoff();
 
+	dfetch_start_system_call();
```

Comment:
If we think of user_exit_irqoff(), dfetch_start_system_call() and trace_hardirqs_off_finish() as opening brackets, and dfetch_end_system_call(), trace_hardirqs_on_prepare(), user_enter_irqoff() as closing brackets, ranges defined by these brackets should be properly nested.
I suggest moving dfetch_start_system_call to the end of enter_from_user_mode() to achieve this.